### PR TITLE
Split maze Subtypes into Subtypes and ResultsHandlers

### PR DESCRIPTION
**Pull request by Hamms**
Where Subtypes preserve all those bits of logic that care about the actual gameplay of the subtype (where you can move, what things you can do, how you manipulate the state of the game) and ResultsHandlers preserve all those bits of logic that care about success/failure handling and feedback (which game states count as passing, what feedback we show based on what game state, etc)

### CODE REVIEW

File: apps/src/maze/bee.js [hunk 1]
```diff
@@ -28,14 +26,13 @@ export default class Bee extends Gatherer {
...
-    this.nectarGoal_ = config.level.nectarGoal || 0;
-    this.honeyGoal_ = config.level.honeyGoal || 0;
-
     // at each location, tracks whether user checked to see if it was a flower or
     // honeycomb using an if block
     this.userChecks_ = [];
 
     this.overrideStepSpeed = 2;
+    this.honey_;
+    this.nectars_;
```
Comment (joshlory) [inline apps/src/maze/bee.js]: Same feedback: let's init these to `undefined` otherwise it's a no-op.

File: apps/src/maze/maze.js [hunk 2]
```diff
@@ -40,6 +42,7 @@ module.exports = class Maze {
 
     this.cachedBlockStates = [];
 
+    this.resultsHandler;
     this.response;
     this.result;
     this.testResults;
```
Comment (joshlory) [inline apps/src/maze/maze.js]: No-op.

File: apps/src/maze/results/utils.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+const FarmerHandler = require('./farmer');
+const BeeHandler = require('./bee');
+const CollectorHandler = require('./collector');
+const WordsearchHandler = require('./wordsearch');
+const HarvesterHandler = require('./harvester');
+const PlanterHandler = require('./planter');
+const ResultsHandler = require('./resultsHandler');
+
+export function createResultsHandlerForSubtype(controller, config) {
```
Comment (joshlory) [inline apps/src/maze/results/utils.js]: I'm not a huge fan of this helper but I can't quite put my finger on why.  Can we use a hash of the subtype name to the appropriate handler, and look up by `controller.subtype.constructor.name`?

Will the extracted repo include `maze/results/*`?

File: apps/src/maze/results/bee.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
+import Gatherer from './gatherer';
+import mazeMsg from '../locale';
+import {
+  TestResults,
+  BeeTerminationValue as TerminationValue
+} from '../../constants.js';
+
+export default class Bee extends Gatherer {
```
Comment (joshlory) [inline apps/src/maze/results/bee.js]: Let's export `BeeHandler` here, to match how it's used in `maze/results/utils.js`.

**Comment by Hamms** [comment]
The idea is that `/maze/results` is not going to be extracted, but all the subtypes themselves are.

The idea is that results is _looking_ into a lot of maze state, because that's where we have the concept of success or failure as it relates to maze state. But it should not in any way be _modifying_ maze state

File: apps/src/maze/results/utils.js (hunk 3 above)
Comment (Hamms) [reply apps/src/maze/results/utils.js]: Sure!

And (as answered in another comment) no it will not

**Comment by Hamms** [comment]
PTAL

**Comment by Hamms** [comment]
I based this split off of the assumption that they were not; if we discover that assumption was false, it shouldn't be too hard to do some rearranging afterwards

**Pull request merged**